UnchainedLauncher.Core/Processes/Chivalry/UnchainedChivalry2Launcher.cs

```diff
@@ -32,34 +53,137 @@ public Either<ProcessLaunchFailure, Process> Launch(ModdedLaunchOptions launchOp
 
             moddedLaunchArgs.Insert(offsetIndex, " " + launchOpts);
 
-            if (!moddedLaunchArgs.Contains("--saveddirsuffix"))
-                moddedLaunchArgs += " --saveddirsuffix=Unchained";
-
             PrepareModdedLaunchSigs();
+            var updateTask = PrepareUnchainedLaunch(updateUnchainedDependencies);
+            updateTask.Wait();
+            var updateResult = updateTask.Result;
+
+            if (updateResult == false) {
+                logger.Error("Failed to launch modded game.");
+            }
+            
 
             logger.Info($"Launch args: {moddedLaunchArgs}");
 
             var launchResult = Launcher.Launch(Path.Combine(InstallationRootDir, FilePaths.BinDir), moddedLaunchArgs);
 
             return launchResult.Match(
                 Left: error => {
-                    error.Match(
-                        LaunchFailedError: e =>
-                            logger.Error($"Failed to launch Chivalry 2 Unchained. {e.ExecutablePath} {e.Args}",
-                                e.Underlying),
-                        InjectionFailedError: e =>
-                            logger.Error($"Failed to inject DLLs into Chivalry 2 Unchained. {e.DllPaths}", e.Underlying)
-                    );
-                    return Prelude.Left(error);
+                    logger.Error(error);
+                    return Left(UnchainedLaunchFailure.LaunchFailed(error));
                 },
                 Right: proc => {
                     logger.Info("Successfully launched Chivalry 2 Unchained");
                     return InjectDLLs(proc);
                 }
             );
         }
+        
+        private async Task<bool> PrepareUnchainedLaunch(bool updateDependencies) {
```

Comment:
This function largely came from the LauncherViewModel. Some additions were made to support updating the unchained mods pak